pallets/rewards/src/migrations.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use frame_support::traits::{
+	fungibles::{Inspect, Mutate},
+	Get, OnRuntimeUpgrade,
+};
+use sp_runtime::traits::AccountIdConversion;
+use sp_std::marker::PhantomData;
+#[cfg(feature = "try-runtime")]
+use {frame_support::inherent::Vec, sp_std::vec};
+
+use crate::{BalanceOf, Config};
+
+pub struct FundExistentialDeposit<T, I, Currency, ED>(PhantomData<(T, I, Currency, ED)>);
+
+fn pallet_is_funded<T, I, Currency, ED>() -> bool
+where
+	T: Config<I>,
+	I: 'static,
+	Currency: Get<<T as Config<I>>::CurrencyId>,
+	ED: Get<BalanceOf<T, I>>,
+{
+	let balance = <<T as Config<I>>::Currency as Inspect<T::AccountId>>::balance(
+		Currency::get(),
+		&<T as Config<I>>::PalletId::get().into_account_truncating(),
+	);
+	balance >= ED::get()
+}
+
+impl<T, I, Currency, ED> OnRuntimeUpgrade for FundExistentialDeposit<T, I, Currency, ED>
+where
+	T: frame_system::Config + Config<I>,
+	I: 'static,
+	Currency: Get<<T as Config<I>>::CurrencyId>,
+	ED: Get<BalanceOf<T, I>>,
+{
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		assert!(!pallet_is_funded::<T, I, Currency, ED>());
```

Comment:
Should it be an `ensure!` instead in order to avoid panicking the whole try runtime execution?

Reply:
Will do. On a note, in Substrate there are migrations with `assert` as well. Luckily, in the future `try-runtime` does not bail on the first error anymore[ according to this PR](https://github.com/paritytech/substrate/pull/14083)  🚀